Tests/CompatibilitySession.Tests.ps1

```diff
@@ -0,0 +1,197 @@
+        $first | Should -Be $second
+    }
+
+    It "Get-WinModule should return a single value for PnpDevice" {
+        $info = Get-WinModule PnpDevice
+        $info | Should -Not -BeNullOrEmpty
+        $info.Name | Should -BeExactly "PnpDevice"
+    }
+
+    It "Import-WinModule should import the PnpDevice module" {
+        # Remove the PnpDevice module if it's installed
+        Get-Module PnpDevice | Remove-Module
+
+        # Now import the proxy module, returning the ModuleInfo object
+        $info = Import-WinModule PnpDevice -PassThru
+        $info | Should -Not -BeNullOrEmpty
+        $info.Name | Should -BeExactly "PnpDevice"
+        $info.ModuleType | Should -BeExactly "Script"
+
+        # Verify that the commands were imported as proxies
+        $cmd = Get-Command Get-PnpDevice
+        $cmd.CommandType | Should -BeExactly "Function"
+
+        # Make sure the command actually runs
+        $results = Get-PnpDevice
+        $results | Should -Not -BeNullOrEmpty
+        $results | Should -BeOfType ([Microsoft.Management.Infrastructure.CimInstance])
+        $results[0].PSObject.TypeNames[0] | Should -BeExactly 'Microsoft.Management.Infrastructure.CimInstance#ROOT/cimv2/Win32_PnPEntity'
+
+        # Clean up
+        Get-Module PnpDevice | Remove-Module
+    }
+
+    It "Import-WinModule Microsoft.PowerShell.Management should import new commands but not overwrite existing ones" {
+        # Remove the proxy module if present
+        Get-Module Microsoft.PowerShell.Management |
+            Where-Object ModuleType -eq Script |
+                Remove-Module
+        # Import the proxy module
+        Import-WinModule  Microsoft.PowerShell.Management
+        $cmd = Get-Command Get-ChildItem
+        $cmd.Module.ModuleType | Should -BeExactly "Manifest"
+        $cmd2 = Get-Command Get-EventLog
+        $cmd2.Module.ModuleType | Should -BeExactly "Script"
+        Get-EventLog -LogName Application -Newest 1 | Should -Not -BeNullOrEmpty
+    }
+
+    It "Add-WinFunction should define a function in the current session and return information from the compatibility session" {
+        Remove-Item -ErrorAction Ignore function:myFunction
+        Add-WinFunction myFunction {param ($n) "Hi $n!"; $PSVersionTable.PSEdition }
+        $function:myFunction | Should -Not -BeNullOrEmpty
+        $msg, $edition = myFunction Bill
+        $msg | Should -BeExactly "Hi Bill!"
+        $edition | Should -BeExactly "Desktop"
+    }
+
+    It "Invoke-WinCommand should return information from the compatibility session" {
+        $result = Invoke-WinCommand { $PSVersionTable.PSEdition }
+        $result | Should -BeExactly "Desktop"
+    }
+
+    It "Compare-WinModule should return a non-null collection of modules" {
+        $pattern = 'Azure*'
+        $modules = Compare-WinModule $pattern
+        $modules | Should -Not -BeNullOrEmpty
+        # Verify that the deserialized object's typenames contains PSModuleInfo
```

Comment:
This comment doesn't match the test. is this  a To-do?

Reply:
No - it's a bad comment, I've removed it.